vscode/src/data/virtualStorage.ts

```diff
@@ -75,12 +75,19 @@ export const writeSolutionsToMemFs = async (
     memFs.createDirectoriesIfNeeded(modifiedUri, KONVEYOR_SCHEME),
   );
 
-  const writeDiff = async ({ diff, originalUri, modifiedUri }: LocalChange) => {
-    const content = await applyDiff(originalUri, diff);
-    memFs.writeFile(modifiedUri, Buffer.from(content), {
-      create: true,
-      overwrite: true,
-    });
+  const writeDiff = async (change: LocalChange) => {
+    const { originalUri, modifiedUri, diff, state } = change;
+    if (state === "applied") {
+      const msg = `Already applied solution for ${originalUri.path}.`;
+      window.showErrorMessage(msg);
```

Comment:
I don't know if an error should be here, I think a log message, but if it is already applied then good things are happening not bad IMO.

Reply:
I think `window.showInformationMessage` is a better choice.

https://code.visualstudio.com/api/references/vscode-api#window.showInformationMessage